### ISSUE TITLE
Support for nested arrays

### ISSUE THREAD

**Issue by derwaldgeist** [aldeed/meteor-simple-schema#529]
I would like to store a GeoJSON Polygon in the database:
https://docs.mongodb.org/v3.0/reference/geojson/#geojson-polygon

According to these docs, the coordinates are a nested array, which would translate to SimpleSchema like this:

```js
var geoJSONPolygonSchema = new SimpleSchema({
  type: {
    type: String,
    allowedValues: ['Polygon']
  },
  coordinates: {
    type: [[[Number]]],
    decimal:true
  }
});
```
Though this is accepted by the validator, all of the coordinates become "null". I tried to use a custom validator, but also in this case the ```this.value``` property of the validation function only sees null values.

**Comment by sebakerckhof** [comment]
Might be related to this: https://github.com/aldeed/meteor-simple-schema/issues/515

In short: this seems fixed in the development version, however, you won't be able to use the '[[]]' syntax, but rather have to specify all array levels using the dollar-sign notation.

**Comment by derwaldgeist** [comment]
Thanks, so I am looking forward for these fixes in the production version!

**Comment by aldeed** [comment]
Yes, I'm still working on a few more big changes and refactors for 2.0, but all known issues are already fixed in 2.0. Hopefully released in a few weeks.

**Comment by derwaldgeist** [comment]
Thanks for your work, highly appreciated! I am closing this now.
